LeaderboardBackend/Services/Impl/AccountConfirmationService.cs

```diff
@@ -48,18 +49,18 @@ public async Task<CreateConfirmationResult> CreateConfirmationAndSendEmail(User
 
         _applicationContext.AccountConfirmations.Add(newConfirmation);
 
-        await _applicationContext.SaveChangesAsync();
-
         try
         {
             await _emailSender.EnqueueEmailAsync(
                 user.Email,
                 "Confirm Your Account",
                 GenerateAccountConfirmationEmailBody(user, newConfirmation)
             );
+            await _applicationContext.SaveChangesAsync();
         }
         catch
         {
+            _applicationContext.AccountConfirmations.Entry(newConfirmation).State = EntityState.Detached;
```

Comment:
I don't think this is necessary. If the call to `SaveChangesAsync` never happens, then the changes should be dropped when the context is disposed ed, which should happen automatically when the thread handling this query terminates.

Reply:
Alternatively, you can store the return value of calling `Add` in a local var and then use that. Or just call `Clear` on the change tracker.

Reply:
See I'm not sure if the query will terminate if I don't change the entity state (by setting it to `Detached` or calling `Clear` on the change tracker). Do we know for sure that it will?

Anyway, as we've discussed in the thread above, I'll just revert this commit entirely.